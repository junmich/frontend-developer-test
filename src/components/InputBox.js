import React from 'react';
import { MdEmail } from 'react-icons/md'

const InputBox = (props) => {
    return (
        <div className="input-box">
            <p className="input-box__icon">{props.icon}</p>
            <input
                type={props.type}
                className="form-control input-box__textbox"
                placeholder={props.placeholder}
                onChange={props.onChange}
                required
                value={props.value || ''}
                autoFocus
                name={props.name || ''}
            />
        </div>
    )
}

InputBox.defaultProps = {
    type: "text",
    placeholder: 'Email Address',
    icon: <MdEmail size={25} />,
onChange: () => { /* do nothing */ }
}
export default InputBox;