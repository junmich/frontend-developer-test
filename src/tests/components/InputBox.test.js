import React from 'react';
import { shallow } from 'enzyme';
import InputBox from '../../components/InputBox';

test('should render Input box correctly', () => {
    const wrapper = shallow(<InputBox />);
    expect(wrapper).toMatchSnapshot();
});

test('should render Input box correctly with props', () => {
    const wrapper = shallow(<InputBox placeholder="Email" />);
    expect(wrapper).toMatchSnapshot();
});


test('should render Input box correctly with props and set name', () => {
    const wrapper = shallow(<InputBox placeholder="Email" name="email" />);
    expect(wrapper).toMatchSnapshot();
});