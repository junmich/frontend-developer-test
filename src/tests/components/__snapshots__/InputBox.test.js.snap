// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input box correctly 1`] = `
<div
  className="input-box"
>
  <p
    className="input-box__icon"
  >
    <MdEmail
      size={30}
    />
  </p>
  <input
    autoFocus={true}
    className="form-control input-box__textbox"
    name=""
    onChange={[Function]}
    placeholder="Email Address"
    required={true}
    type="text"
    value=""
  />
</div>
`;

exports[`should render Input box correctly with props 1`] = `
<div
  className="input-box"
>
  <p
    className="input-box__icon"
  >
    <MdEmail
      size={30}
    />
  </p>
  <input
    autoFocus={true}
    className="form-control input-box__textbox"
    name=""
    onChange={[Function]}
    placeholder="Email"
    required={true}
    type="text"
    value=""
  />
</div>
`;

exports[`should render Input box correctly with props and set name 1`] = `
<div
  className="input-box"
>
  <p
    className="input-box__icon"
  >
    <MdEmail
      size={30}
    />
  </p>
  <input
    autoFocus={true}
    className="form-control input-box__textbox"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    required={true}
    type="text"
    value=""
  />
</div>
`;
